Clarification of CNF Conformance Program, adding CNF Working Group info (#450)
cncf/cnf-conformance: README.md

# CNF Conformance

> The goal of the CNF Conformance Program is to provide an open source test suite to demonstrate conformance and implementation of cloud native best practices for both open and closed source Cloud Native Network Functions. The conformance program is a living thing. The CNCF community, through the [Cloud Native Network Function Working Group](cnf-wg/README.md), oversees and maintains what it means to be a cloud native conformant telco application (including those applications called network functions). It also develops the process and policy around the certification program. Work on the mechanics of the conformance tests occurs in the [CNF Conformance Test Suite](README-testsuite.md).

Reviewer: just the test suite? or does the conformance umbrella provide other things?

Author: ATM the Program consists of the WG and the test suite project.

- The WG will produce requirements and best practice information for CNFs. It will also create the process for certifying.
- The Test suite project will create tests and maintain the testing software.

No other things as of yet.

Reviewer: only reason I asked as its phrased as the CNF Conformance Program as opposed to specifically referencing the Test Suite. Just confused me.

Reviewer 2: I'm good with that definition

Author: @jeffsaelens the top level REAMDE is for the umbrella program which includes a test suite project and the new working group. In the future we may split things into different repos.

## Why Conformance Matters

> With such a wide array of applications being developed today, workload conformance tests help ensure that developers can follow cloud native best practices when building greenfield applications and/or modernizing existing applications. A conformance passing application provides the following guarantees:

Reviewer 3: I think the last sentence is not needed.

Reviewer 4: which sentence are you referring to?

Best practices: Your application follows cloud native best practices. This is useful to know whether you are building upon the work of the community or handling your own custom setup.

Predictability: Your application acts in a predictable manner when running on cloud native infrastructure like Kubernetes. Unexpected behavior should be rare, because application specific issues are weeded out during the conformance tests.

> Interoperability: Workloads can be ported across various cloud native infrastructures. This standardization is a key advantage of open source software, and allows you to avoid vendor lock-in.

Reviewer 5: There was a question on today's TUG call in the lines of "can my CNF depend on an SDN infrastructure for e.g. SFC?". Apparently, this is an important need for some of the solutions that do exist, which essentially opens the following questions:
 - how "smart/programmable" is the cloud-native infrastructure expected to be?
 - shall we consider the "smartness" being an application that runs on top of the "dumb" infrastructure?
 - if I depend on that "smartness" do I bring it with my app, or it's better to just drop and rethink my app

Author: IMO, the application should degrade gracefully.  Design to expect failure. Ideally the app can provide a degraded service even if piece of hardware is not available.

A new workload can describe what it wants and ideally orchestrator will try to match a node which has those resources. This would be similar to asking for a system with a GPU for processing or something else.

Reviewer 2: In the case of SFC which is a global construct, a CNF is but one component in the service graph. Question would then be wether the SFC control (understanding relations between CNFs) should it be part of the Conformance scenario or independent.

> Running applications in a cloud native manner will allow you more fully utilize the advantages of cloud native infrastructure. The CNF Conformance program enables interoperability of Cloud native Network Functions (CNFs) from multiple vendors running on top of Kubernetes.

Reviewer 2: Should it say "Running applications in a cloud native manner will allow for fully benefit from the advantages ..."

Author: ```suggestion
Implementing and running applications in a cloud native manner will enable you to more fully benefit from the advantages cloud native infrastructure.
```

Reviewer 2: 👍

Reviewer 6: 👍
